api/src/handlers/websocket/main.go

```diff
@@ -0,0 +1,72 @@
+package websocket
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/gorilla/websocket"
+	"net/http"
+	"encoding/json"
+	"time"
+	"porter/api/src/lib/github"
+	"porter/api/src/handlers/websocket/message"
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+}
+
+type ReceivedSocketEvent struct {
+	Type string `json:"type"`
+}
+
+func Handler(c *gin.Context) {
+	conn, err := upgrader.Upgrade(c.Writer, c.Request, nil)
```

Comment:
whats the idea behind making this websockets? 😉

Reply:
This process can take a while since there are many requests, I wanna add a log of all the action in the client so it will be nicer to see what's happening (like terminal).
so I'm opening a socket connection and pushing messages to display them in the client.